etl/assets.py

```diff
@@ -153,3 +146,19 @@ def wikipedia_anti_recommendations_json_file(
             json.dumps(anti_recommendation_graph)
             for anti_recommendation_graph in wikipedia_anti_recommendations.anti_recommendation_graphs
         )
+
+
+@asset
+def wikipedia_arkg(
+    wikipedia_anti_recommendations: AntiRecommendationGraphTuple,
+    arkg_config: ArkgConfig,
+    output_config: OutputConfig,
+) -> None:
```

Comment:
Put the Path here, that way downstream assets don't have to know about OutputConfig.

Similar with the embedding story. The generic pattern here is to pass a lightweight "handle" or "descriptor" downstream since you can't pass the entire database. The descriptor could be a path or e.g., an S3 key.

Reply:
In my Dagster code I make the descriptors opaque. Something like this.

```
class Database:
    @dataclass(frozen=True)
    class Descriptor:
        path: Path  # No one should access this except Database!

   def __init__(self, ...private constructor params...): 

   @classmethod
   def create(parameters_from_asset) -> Database:
       pass

   def use_database_heavyweight(): pass

   @property
   def descriptor(self) -> Descriptor:
       return Descriptor(self.__internal_path)
 
   @classmethod
   def open(descriptor: Descriptor) -> Database:
        # Then only Database knows what to do with descriptor

@asset
def database() -> Database.Descriptor:
    database = Database.create()
    return database.descriptor

@asset
def downstream_asset(database: Database.Descriptor):
    # Don't know and don't care what's in the descriptor
    database_ = Database.open(database)
    ...
```

This is the same pattern as file descriptors: https://en.wikipedia.org/wiki/File_descriptor

Reply:
Very cool. I created `ArkgDatabase` and `EmbeddingDatabase` to pass descriptors.

Reply:
They both have `write` methods, but they return different types so I didn't make it an abstract method.